terraform/modules/cloudwatch/main.tf

```diff
@@ -0,0 +1,49 @@
+resource "aws_cloudwatch_event_rule" "ecs_scheduled_event" {
+  name        = local.cloudwatch_event_rule_name
+  description = "Scheduled event rule to run ECS task"
+  schedule_expression = var.cron_expression
+}
+
+resource "aws_cloudwatch_event_target" "ecs_event_target" {
+  rule      = aws_cloudwatch_event_rule.ecs_scheduled_event.name
+  target_id = "run-ecs-task"
+
+  arn = data.aws_ecs_cluster.existing_cluster.arn
+
+  ecs_target {
+    task_count          = 1
+    launch_type         = var.launch_type
+    platform_version    = "LATEST"
+    task_definition_arn = data.aws_ecs_task_definition.existing_task.arn
+  }
+}
+
+resource "aws_sns_topic" "slack_notifications" {
+  name = local.sns_topic_name
+}
+
+resource "aws_sns_topic_subscription" "slack_subscription" {
+  topic_arn = aws_sns_topic.slack_notifications.arn
+  protocol  = "https"
+  endpoint  = var.slack_notification_endpoint
+}
+
+resource "aws_iam_role" "cloudwatch_events_role" {
```

Comment:
can the IAM resources be split out into an "iam.tf" file - this makes it easier to review these if we are asked for this by cloudone